src/Model/Contact/User.php

```diff
@@ -21,14 +21,56 @@
 
 namespace Friendica\Model\Contact;
 
+use Friendica\Core\Logger;
+use Friendica\Core\System;
+use Friendica\Database\Database;
 use Friendica\Database\DBA;
 use Friendica\Model\Contact;
+use Friendica\Model\ItemURI;
 
 /**
  * This class provides information about user related contacts based on the "user-contact" table.
  */
 class User
 {
+	/**
+	 * Insert a user-contact for a given contact array
+	 *
+	 * @param array $contact
+	 * @return void
+	 */
+	public static function insertForContactArray(array $contact)
```

Comment:
Isn't this going to create entries in `user-contact` with `uid = 0`?

Reply:
Yes. I thought about it for a while. When we want to use this table as the leading table for a future view then it has to contain all contacts - like the `post-user` table also contains entries for `uid=0`. When we realize that this was a bad idea, then we can always delete it.

Reply:
It just doesn't make any sense. Either we're querying contacts the user has ever interacted with (with `contact-user` as the leading table`) or we're querying a public contact record (from `contact`). It doesn't make sense to use `contact-user` for everything.

Reply:
It will all depend on the upcoming query. And like I said: Possibly we will need this data. If not, then we can delete with no problem.

Reply:
Isn't it easier to plan ahead and not have to remove something that was wrongly added?

Reply:
The removal is one line of code. If we really need all this data is something that we will only see during implementation.

Reply:
You're the boss, boss.